lib/inc/facter/facts/collection.hpp

```diff
@@ -129,6 +129,14 @@ namespace facter { namespace facts {
         // Visual Studio 12 still doesn't allow default for move assignment.
         collection& operator=(collection&& other);
 
+        /**
+         * Gets the blocklist associated with the collection.
+         * @return Returns set of blocked groups.
+         */
+        std::set<std::string> blocklist(){
```

Comment:
header should contain only declaration  `std::set<std::string> blocklist() const;`, while corresponding cpp file should contain the definition.
also `const` whould be used if class is not changed